plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java

```diff
@@ -0,0 +1,299 @@
+    {
+        String schemaName = "test_basic_operations_schema_" + randomNameSuffix();
+        String schemaLocation = locationPattern.formatted(bucketName, schemaName, schemaName);
+        String tableName = "test_basic_operations_table_" + randomNameSuffix();
+        String qualifiedTableName = schemaName + "." + tableName;
+        String partitionQueryPart = (partitioned ? " WITH (partitioned_by = ARRAY['col_int'])" : "");
+
+        assertUpdate("CREATE SCHEMA " + schemaName + " WITH (location = '" + schemaLocation + "')");
+        assertThat(getSchemaLocation(schemaName)).isEqualTo(schemaLocation);
+
+        assertUpdate("CREATE TABLE " + qualifiedTableName + "(col_str varchar, col_int int)" + partitionQueryPart);
+        String expectedTableLocation = (schemaLocation.endsWith("/") ? schemaLocation : schemaLocation + "/") + tableName;
+        expectedTableLocation = adaptTableLocation(expectedTableLocation);
+
+        String actualTableLocation = metastore.getTable(schemaName, tableName).orElseThrow().getStorage().getLocation();
+        assertThat(actualTableLocation).matches(expectedTableLocation);
+
+        assertUpdate("INSERT INTO " + qualifiedTableName + "  VALUES ('str1', 1), ('str2', 2), ('str3', 3)", 3);
+        assertQuery("SELECT * FROM " + qualifiedTableName, "VALUES ('str1', 1), ('str2', 2), ('str3', 3)");
+
+        assertThat(getTableFiles(actualTableLocation)).isNotEmpty();
+        validateDataFiles(partitioned ? "col_int" : "", qualifiedTableName, actualTableLocation);
+
+        assertUpdate("DROP TABLE " + qualifiedTableName);
+        assertThat(getTableFiles(actualTableLocation)).isEmpty();
+
+        assertUpdate("DROP SCHEMA " + schemaName);
+        validateFilesAfterDrop(actualTableLocation);
+    }
+
+    @Override
+    @Test(dataProvider = "locationPatternsDataProvider")
+    public void testMergeWithProvidedTableLocation(boolean partitioned, String locationPattern)
+    {
+        // Row-level modifications are not supported for Hive tables
+    }
+
+    @Test(dataProvider = "locationPatternsDataProvider")
+    public void testAnalyzeWithProvidedTableLocation(boolean partitioned, String locationPattern)
+    {
+        String tableName = "test_analyze_" + randomNameSuffix();
+        String location = locationPattern.formatted(bucketName, schemaName, tableName);
+        String partitionQueryPart = (partitioned ? ",partitioned_by = ARRAY['col_int']" : "");
+
+        assertUpdate("CREATE TABLE " + tableName + "(col_str, col_int)" +
+                "WITH (external_location = '" + location + "'" + partitionQueryPart + ") " +
+                "AS VALUES ('str1', 1), ('str2', 2), ('str3', 3)", 3);
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('str4', 4)", 1);
+        assertQuery("SELECT * FROM " + tableName, "VALUES ('str1', 1), ('str2', 2), ('str3', 3), ('str4', 4)");
+
+        //Check statistics collection on write
+        if (partitioned) {
```

Comment:
Shouldn't we separate cases for partitioned and not partitioned. I don't like idea that test flow depends on the test arguments.